include/xsimd/types/xsimd_batch.hpp

```diff
@@ -451,14 +437,15 @@ namespace xsimd
     }
 
     /**
-     * Create a batch with elements initialized from \c data.
-     * It is an error to have `data.size() != size.
+     * Create a batch with elements initialized from \c val0, \c val1, \c vals...
+     * There must be exactly \c size elements in total.
      */
     template <class T, class A>
-    inline batch<T, A>::batch(std::initializer_list<T> data) noexcept
-        : batch(data.begin(), detail::make_index_sequence<size>())
+    template <class... Ts>
+    inline batch<T, A>::batch(T val0, T val1, Ts... vals) noexcept
+        : batch(kernel::set<A>(batch {}, A {}, val0, val1, static_cast<T>(vals)...))
     {
-        details::check_batch_init(*this, data);
+        static_assert(sizeof...(Ts) + 2 == size, "as many arguments as batch elements");
```

Comment:
Maybe the static assert message should say something like `"The constructor requires as many arguments as batch elements."`

Reply:
Yup, this would be only note too. Looks great to me otherwise!